pkg/resource/target_reconciler.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resource
+
+import (
+	"context"
+	"time"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/crossplaneio/crossplane-runtime/apis/core/v1alpha1"
+	"github.com/crossplaneio/crossplane-runtime/pkg/logging"
+	"github.com/crossplaneio/crossplane-runtime/pkg/meta"
+)
+
+const (
+	targetControllerName   = "kubernetestarget.crossplane.io"
+	targetReconcileTimeout = 1 * time.Minute
+
+	errGetTarget                 = "unable to get Target"
+	errMissingSecretRef          = "no secretRef specified for Target with a clusterRef"
+	errManagedResourceIsNotBound = "managed resource in Target clusterRef is unbound"
+	errUpdateTarget              = "unable to update Target"
+)
+
+// A TargetKind contains the type metadata for a kind of target resource.
+type TargetKind schema.GroupVersionKind
+
+// A TargetReconciler reconciles targets by propagating the secret of the
+// referenced managed resource.
+type TargetReconciler struct {
+	client     client.Client
+	newTarget  func() Target
+	newManaged func() Managed
+
+	propagator ManagedConnectionPropagator
+}
+
+// NewTargetReconciler returns a Reconciler that reconciles
+// KubernetesTarget by setting their ConnectionSecretRef to that of the
+// referenced Kubernetes cluster implementation.
```

Comment:
This reads a little like we literally use the referenced cluster's secret, rather than propagating it.

Reply:
Agree here, will modify.